src/core/model_repository_manager.cc

```diff
@@ -280,7 +280,7 @@ GetModifiedTime(const std::string& path)
 
   // If 'path' is a file return its mtime. Otherwise, using the modification
   // time of the directory as baseline in case of file deletion
-  int64_t mtime;
+  int64_t mtime = 0;
   status = FileModificationTime(path, &mtime);
```

Comment:
We need to fix the definition of FileModificationTime. If it returns success it should return a valid file modification time. If we ask for modification time for a file/folder and none is available we should have some defined behavior, like setting the modification time to 0 or returning a specific error code.

Reply:
I think we need not return an error for S3 and other filesystems that do not have a concept of timestamps for folders. The logic inside GetModifiedTime would log an unnecessary error in such a case. I'd rather we return 0 in the case where the path corresponds to a directory which does not have a modification time. 

That being said 0 is still not a valid file modification time.